### PR TITLE
handle freebsd correctly

### PR DESCRIPTION
**Pull request by thejambavan**
#### This Pull Request (PR) fixes the following issues

Fails to install or configure on FreeBSD due to lack of systemd

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -7,10 +6,28 @@
 #
 class mosquitto::config (
   Array[String[1]] $config = $mosquitto::config,
+  $prefix                  = undef
 ) {
   assert_private()
+  # Sensible defaults for the config prefix if none is specified
+  unless $prefix{
+    case $facts['os']['name'] {
```
Comment (ekohl) [inline manifests/config.pp:14]: Wouldn't it be better to use the kernel fact?

File: manifests/config.pp [hunk 2]
```diff
@@ -7,10 +6,28 @@
 #
 class mosquitto::config (
   Array[String[1]] $config = $mosquitto::config,
+  $prefix                  = undef
 ) {
   assert_private()
+  # Sensible defaults for the config prefix if none is specified
+  unless $prefix{
```
Comment (ekohl) [inline manifests/config.pp:13]: I'd expect some if/else where the prefix is used. Now it's just behaving as a boolean.

File: manifests/config.pp [hunk 3]
```diff
@@ -7,10 +6,28 @@
 #
 class mosquitto::config (
   Array[String[1]] $config = $mosquitto::config,
+  $prefix                  = undef
```
Comment (ekohl) [inline manifests/config.pp:9]: Could you ensure there's a data type?

**Comment by evgeni** [comment]
Hi, #32 was merged and should have everything you need. Mind checking?

**Comment by ekohl** [comment]
I'm going to assume #32 was sufficient.

**Pull request closed without merging**